packages/patterns/docs/marshal-vs-patterns-level.md

```diff
@@ -0,0 +1,94 @@
+|              | `copyArray`  | `copyArray`      | sequence of passable values    |
+|              | `copyRecord` | `copyRecord`     | set of name,passable pairs     |
+|              | `copySet`    | `tagged`         | Set of unique keys             |
+|              | `copyBag`    | `tagged`         | Multiset of key,count pairs    |
+|              | `copyMap`    | `tagged`         | Map from keys to passables     |
+| Matchers     | `match:*`    | `tagged`         | Non-literal patterns           |
+| Guards (TBD) | `guard:*`    | `tagged`         | Non-pattern guards             |
+| Just Tagged  | undefined    | `tagged`         | Not understood to have a kind  |
+| Capability   |              |                  |                                |
+|              | `remotable`  | `remotable`      | behavior + identity            |
+|              | `promise`    | `promise`        | Promise                        |
+| Other        |              |                  |                                |
+|              | `error`      | `error`          | best efforts loose diagnostics |
+
+
+## Data, Keys, and Patterns
+
+At the `passStyleOf` level, the containers are CopyArray, CopyRecord, and Tagged. Any passable value is a possibly-empty tree of containers ending in non-container leaves. If the container tree is empty, then the passable value is some non-container value as the leaf of the empty tree. If the leaves do not contain Capabilities, i.e., Remotables or Promises, then the Passable value is Data. It carries only immutable information, without either connection to anything external, nor any unforgeable identity.
+
+Guards do not yet exist as distinct kinds, so we ignore them for now. TODO: Expand this if kinds expand to include guards.
+
+At the `kindOf` level, the containers are CopyArray, CopyRecord, CopySet, CopyBag, and CopyMap. At this level, Matchers, possibly Guards, and unrecognized Taggeds are not considered containers. Again, Any passable value is a possibly-empty tree of containers ending in non-container leaves. The containers at the `passStyleOf` level, CopyArray, CopyRecord, and Tagged, do not need any comparability among their elements.
+
+By contrast, the elements of CopySets and CopyBags, as well as the keys of CopyMaps must be comparable for equality. This subset of passable values are the Keys. Because these containers are pass-by-copy, this equality must be pass-invariant: If passing xa from vatA to vatB arrives as xb, and likewise ya and yb, then `sameKey(xa,ya)` iff `sameKey(xb,yb)`. We do not wish to give Promises, Errors, or unrecognized Taggeds any useful pass-invariant equality. Thus, the leaves of a Key's kind-containment tree may not include Promises, Errors, or unrecognized Taggeds.
+
+These conditions all apply to Patterns as well. The differences are
+   * Patterns can contain Matchers, but Keys cannot. Thus, all Keys are Patterns, but Patterns that include Matchers are not Keys.
```

Comment:
We should find somewhere to put in something like "Primitives ⊂ Scalars ⊂ Data ⊂ Keys ⊂ Patterns ⊂ Passables", textually and/or graphically. And maybe even more definitively:
* Passables = Containable\<Keys, Patterns ∪ Promises ∪ Errors ∪ UnrecognizedTaggeds>
* Patterns = Containable\<Keys, Keys ∪ Matchers>
* Keys = Containable\<Scalars\>
* Data = Containable\<Primitives>
* Scalars = Primitives ∪ Capabilities
* Capabilities = Remotables ∪ Promises
* Containable\<_K_> = Containable\<_K_, _K_>
* Containable\<_K_, _V_> = Containable<sub>∞</sub>\<_K_, _V_>
* Containable₀\<_K_, _V_> = _K_ ∪ _V_
* Containableₙ₊₁\<_K_, _V_> = Containableₙ\<_K_, _V_> ∪ (**C**\<Containableₙ\<_K_, _V_>\> where **C** is CopyArray or CopyRecord) ∪ (**C**\<Containableₙ\<_K_, _K_>> where **C** is CopySet or CopyBag) ∪ CopyMap<Containableₙ\<_K_, _K_>, Containableₙ\<_K_, _V_>>

Reply:
Great suggestions! Please do!